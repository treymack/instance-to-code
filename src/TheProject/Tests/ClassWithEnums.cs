using System;
using System.Linq;
using ApprovalTests;
using NUnit.Framework;

namespace TheProject.Tests
{
    [TestFixture]
    public class ClassWithEnums
    {
        [Test]
        public void CanScriptToCode()
        {
            var instance = new EnumClass
            {
                Enum1 = Enum1.Value1,
                Enum2 = Enum2.Value2,
            };

            var code = ITC.Instance.ToCode(instance);

            Approvals.Verify(code);
        }
    }

    public enum Enum2
    {
        Value1,
        Value2,
    }

    public enum Enum1
    {
        Value1,
    }

    public class EnumClass
    {
        public Enum1 Enum1 { get; set; }
        public Enum2 Enum2 { get; set; }
    }
}
