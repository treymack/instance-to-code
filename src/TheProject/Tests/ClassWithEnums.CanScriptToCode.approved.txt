new TheProject.Tests.EnumClass()
{
Enum1 = TheProject.Tests.Enum1.Value1,
Enum2 = TheProject.Tests.Enum2.Value2,
}